protocol/shannon/mode_delegated.go

```diff
@@ -33,16 +39,25 @@ func (p *Protocol) getDelegatedGatewayModeApps(ctx context.Context, httpReq *htt
 
 	logger.Debug().Msgf("fetching the app with the selected address %s.", selectedAppAddr)
 
-	selectedApp, err := p.FullNode.GetApp(ctx, selectedAppAddr)
+	selectedSession, err := p.FullNode.GetSession(ctx, serviceID, selectedAppAddr)
```

Comment:
Not a blocker, but needs a TODO: this will miss the case of app dropping the gateway delegation, and attempts to send a relay anyways.

Reply:
NVM, saw the check below using session's App field.